### PR TITLE
code clean up - debug log format

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#1521

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

We no longer need to do isDebugEnabled check in slf4j.Logger. It is already had performance improvement integrated. 

### Tests

- [X] The following tests are written for this issue:

NA

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-agent/src/main/java/org/apache/helix/agent/AgentStateModel.java [hunk 1]
```diff
@@ -148,8 +148,8 @@ public void genericStateTransitionHandler(Message message, NotificationContext c
     // + " output:\n" + externalCmd.getStringOutput());
 
     if (_logger.isDebugEnabled()) {
```
Comment (desaikomal) [inline helix-agent/src/main/java/org/apache/helix/agent/AgentStateModel.java:150]: you need to remove this line.

File: helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java [hunk 2]
```diff
@@ -130,12 +130,12 @@ public IdealState computeNewIdealState(String resourceName,
         .computePartitionAssignment(allNodes, liveNodes, currentMapping, clusterData);
 
     if (LOG.isDebugEnabled()) {
```
Comment (desaikomal) [inline helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java:132]: this is not required

**Review by desaikomal — COMMENTED** [review]
please fix those 2 things and you are ready for approval... thanks

**Review by desaikomal — APPROVED** [review]

File: helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java [hunk 3]
```diff
@@ -129,14 +129,12 @@ public IdealState computeNewIdealState(String resourceName,
...
 
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("currentMapping: " + currentMapping);
-      LOG.debug("stateCountMap: " + stateCountMap);
-      LOG.debug("liveNodes: " + liveNodes);
-      LOG.debug("allNodes: " + allNodes);
-      LOG.debug("maxPartition: " + maxPartition);
-      LOG.debug("newMapping: " + newMapping);
-    }
+      LOG.debug("currentMapping: {}", currentMapping);
```
Comment (desaikomal) [inline helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java:132]: nit: alignment may be off. please fix it if possible.

File: helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java [hunk 4]
```diff
@@ -60,9 +60,7 @@ public IdealState computeNewIdealState(String resourceName,
 
     IdealState cachedIdealState = getCachedIdealState(resourceName, clusterData);
     if (cachedIdealState != null) {
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("Use cached IdealState for " + resourceName);
-      }
+        LOG.debug("Use cached IdealState for {}", resourceName);
```
Comment (desaikomal) [inline helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java:63]: nit: alignment.

File: helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java [hunk 5]
```diff
@@ -181,16 +179,14 @@ public IdealState computeNewIdealState(String resourceName,
...
-      LOG.debug("currentMapping: " + currentMapping);
-      LOG.debug("stateCountMap: " + stateCountMap);
-      LOG.debug("liveEnabledNodes: " + liveEnabledNodes);
-      LOG.debug("activeNodes: " + activeNodes);
-      LOG.debug("allNodes: " + allNodes);
-      LOG.debug("maxPartition: " + maxPartition);
-      LOG.debug("newIdealMapping: " + newIdealMapping);
-      LOG.debug("finalMapping: " + finalMapping);
-    }
+      LOG.debug("currentMapping: {}", currentMapping);
```
Comment (desaikomal) [inline helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java:182]: nit: alignment

File: helix-core/src/main/java/org/apache/helix/manager/zk/ZkCacheEventThread.java [hunk 6]
```diff
@@ -74,11 +72,9 @@ public void run() {
         } catch (ThreadDeath death) {
           throw death;
         } catch (Throwable e) {
-          LOG.error("Error handling event " + zkEvent, e);
-        }
-        if (LOG.isDebugEnabled()) {
-          LOG.debug("Delivering event #" + eventId + " done");
+          LOG.error("Error handling event {} ", zkEvent, e);
         }
+          LOG.debug("Delivering event # {} done", eventId);
```
Comment (desaikomal) [inline helix-core/src/main/java/org/apache/helix/manager/zk/ZkCacheEventThread.java:77]: nit: alignment

**Review by desaikomal — APPROVED** [review]

**Review by qqu0127 — APPROVED** [review]

File: helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java (hunk 3 above)
Comment (xyuanlu) [reply helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java]: TFTR. Updated

File: helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java [hunk 7]
```diff
@@ -130,23 +130,21 @@ public synchronized void refresh(HelixDataAccessor accessor) {
         }
         _propertyDataChangedMap.put(HelixConstants.ChangeType.CUSTOMIZED_VIEW, false);
       }
-        break;
-      default:
-        break;
+      break;
+        default:
```
Comment (rahulrane50) [inline helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java:134]: nit : can remove this?

File: helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java [hunk 8]
```diff
@@ -130,23 +130,21 @@ public synchronized void refresh(HelixDataAccessor accessor) {
...
+          break;
       }
     }
     long endTime = System.currentTimeMillis();
     LOG.info("END: RoutingDataCache.refresh() for cluster " + _clusterName + ", took " + (endTime
         - startTime) + " ms");
 
-    if (LOG.isDebugEnabled()) {
```
[the remark this answers is not in the document]
Comment (xyuanlu) [reply helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java:142]: TFTR. Good call. Updated.

File: helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java (hunk 7 above)
Comment (xyuanlu) [reply helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java]: Thanks for the suggestion. This is an accidental change and i changed it back. Let's keep this PR only for log formatting.

**Comment by xyuanlu** [comment]
This PR is ready to be merged. Approved by @qqu0127 and @desaikomal

**Pull request merged**